From e90429df534dd39ba17929e5ce6b242fe32a1351 Mon Sep 17 00:00:00 2001
From: Hugo Osvaldo Barrera <hugo@whynothugo.nl>
Date: Mon, 25 Aug 2025 12:35:33 +0200
Subject: [PATCH] tests: upgrade to Qt6

Qt5 is past its EOL and downstream distributions are working on removing
it from their repositories.

This dependency is only used in a test, so it doesn't affect
dependencies or functionality for end users.

Patch-Source: https://github.com/python-lsp/python-lsp-server/pull/667
---
 pyproject.toml                  | 2 +-
 test/plugins/test_completion.py | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 7b0c205..e72c8c8 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -57,7 +57,7 @@ test = [
     "numpy",
     "pandas",
     "matplotlib",
-    "pyqt5",
+    "pyqt6",
     "flaky",
     "websockets>=10.3",
 ]
diff --git a/test/plugins/test_completion.py b/test/plugins/test_completion.py
index 3ba8dbd..015d0c4 100644
--- a/test/plugins/test_completion.py
+++ b/test/plugins/test_completion.py
@@ -282,8 +282,8 @@ def test_jedi_method_completion(config, workspace) -> None:
     reason="Test in Python 3 and not on CIs on Linux because wheels don't work on them.",
 )
 def test_pyqt_completion(config, workspace) -> None:
-    # Over 'QA' in 'from PyQt5.QtWidgets import QApplication'
-    doc_pyqt = "from PyQt5.QtWidgets import QA"
+    # Over 'QA' in 'from PyQt6.QtWidgets import QApplication'
+    doc_pyqt = "from PyQt6.QtWidgets import QA"
     com_position = {"line": 0, "character": len(doc_pyqt)}
     doc = Document(DOC_URI, workspace, doc_pyqt)
     completions = pylsp_jedi_completions(config, doc, com_position)
-- 
2.51.0

